plugins/out_s3/s3.c

```diff
@@ -2235,6 +2317,7 @@ static int cb_s3_exit(void *data, struct flb_config *config)
             ctx->s3_client->upstream->flags &= ~(FLB_IO_ASYNC);
         }
         flb_plg_info(ctx->ins, "Sending all locally buffered data to S3");
+        ctx->has_old_buffers = FLB_TRUE;
```

Comment:
Why set this field here? This field has_old_buffers is for the startup case.

Reply:
To differentiate between each cases, it may be simpler to add a boolean arg to put_all_chunks:

```
put_all_chunks(struct flb_s3 *ctx, int is_startup)
```